EpsilonCCS/Src/CanParsers/BmsCanParser.c

```diff
@@ -0,0 +1,200 @@
+#include "BmsCanParser.h"
+
+#include "BatteryData.h"
+#include "BatteryFaultsData.h"
+
+extern struct BatteryData batteryData;
+extern struct BatteryFaultsData batteryFaultsData;
+
+#define BMU_HEARTBEAT_ID 0x300
+#define STARTUP_INFO_ID 0x301
+#define PACK_INFO_ID 0x302
+#define ERRORS_ID 0x303
+#define TEMPINFO_ID 0x304
+#define CELL_VOLTAGES_ID 0x305
+
+void parseBmuCanMessage(uint32_t stdId, uint8_t data[8])
+{
+    switch (stdId)
+    {
+        case BMU_HEARTBEAT_ID:
+            parseBmuHeartbeat(data);
+            break;
+
+        case STARTUP_INFO_ID:
+            parseStartupInfo(data);
+            break;
+
+        case PACK_INFO_ID:
+            parsePackInfo(data);
+            break;
+
+        case ERRORS_ID:
+            parseErrors(data);
+            break;
+
+        case TEMPINFO_ID:
+            parseTempInfo(data);
+            break;
+
+        case CELL_VOLTAGES_ID:
+            parseCellVoltages(data);
+            break;
+    }
+}
+
+void parseBmuHeartbeat(uint8_t data[8])
+{
+    batteryData.bmuAlive = data[0];
+}
+
+void parseStartupInfo(uint8_t data[8])
+{
+
+    batteryData.bmsRelayStatus = data[0];
+
+    batteryData.populatedCells = data[1];
+
+    uint16_t input12voltInt = // Units 0.1 V
+        (data[2] << 0) +
```

Comment:
Change all instances of + to `|`. This is an OR operation so best to use the OR operator. While using a + is basically the same it is not exactly and can introduce hard to find bugs (mostly related to int overflow and/or negatives).